src/db/models/user.rs

```diff
@@ -137,6 +139,21 @@ impl User {
         matches!(self.api_key, Some(ref api_key) if crate::crypto::ct_eq(api_key, key))
     }
 
+    pub fn set_external_id(&mut self, external_id: Option<String>) {
+        //Check if external id is empty. We don't want to have
+        //empty strings in the database
+        match external_id {
+            Some(external_id) => {
+                if external_id.is_empty() {
+                    self.external_id = None;
+                } else {
+                    self.external_id = Some(external_id)
+                }
+            }
+            None => self.external_id = None,
+        }
```

Comment:
This could be a bit less complex i think. And it will also look a bit more nice.

```suggestion
    let mut ext_id: Option<String> = None;
    if let Some(external_id) = external_id {
        if !external_id.is_empty() {
           ext_id = Some(external_id);
        }
    }
    self.external_id = ext_id;
```